### PR TITLE
implemented article analyzer

### PR DESCRIPTION
**Pull request by Wow11One**
Not sure about architecture, so ready to any сritics

### CODE REVIEW

File: article-analyzer/src/main/java/com/articleanalyzer/core/services/ArticleAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    public ArticleAnalyzer(CategoryService categoryService, UserNotifier userNotifier, CategoryRepository categoryRepository) {
+        this.categoryService = categoryService;
+        this.userNotifier = userNotifier;
+        this.categoryRepository = categoryRepository;
+    }
+
+    public void analyze() {
+        categoryRepository.saveCategories(categoryService.getCategories());
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/com/articleanalyzer/core/services/ArticleAnalyzer.java]:
```suggestion
        var categories = categoryService.getCategories()
        categoryRepository.saveCategories(categories );
```
avoid nested method calls

File: article-analyzer/src/main/java/com/articleanalyzer/core/services/ArticleAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.articleanalyzer.core.services;
+
+import com.articleanalyzer.core.models.User;
+import com.articleanalyzer.core.repository.CategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/com/articleanalyzer/core/services/ArticleAnalyzer.java]: avoid *

File: article-analyzer/src/main/java/com/articleanalyzer/core/repository/UserRepository.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.articleanalyzer.core.models.Article;
+import com.articleanalyzer.core.models.User;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Repository;
+
+import java.io.*;
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/com/articleanalyzer/core/repository/UserRepository.java]: avoid *

File: article-analyzer/src/main/java/com/articleanalyzer/core/services/CategoryService.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package com.articleanalyzer.core.services;
+
+import com.articleanalyzer.core.repository.CategoryRepository;
+import com.google.gson.Gson;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/com/articleanalyzer/core/services/CategoryService.java]: check all classes about *

File: article-analyzer/src/main/java/com/articleanalyzer/core/ArticleAnalyzerApplication.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ApplicationContext;
+
+@SpringBootApplication
+public class ArticleAnalyzerApplication {
+
+	public static void main(String[] args) {
+        ApplicationContext applicationContext = SpringApplication.run(ArticleAnalyzerApplication.class, args);
+        ArticleAnalyzer analyzer = applicationContext.getBean(ArticleAnalyzer.class);
+        analyzer.analyze();
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/com/articleanalyzer/core/ArticleAnalyzerApplication.java:14]: fix formatting

**Review by vitaliiStefanchakInventorsoft — COMMENTED** [review]
nice work! just few small comments, and please commit categories json

File: article-analyzer/src/main/java/com/articleanalyzer/core/ArticleAnalyzerApplication.java (hunk 5 above)
Comment (Wow11One) [reply article-analyzer/src/main/java/com/articleanalyzer/core/ArticleAnalyzerApplication.java:14]: you mean in the whole project?

Comment (Wow11One) [reply article-analyzer/src/main/java/com/articleanalyzer/core/ArticleAnalyzerApplication.java:14]: pushed new commit, added categories.json and fixed some mistakes

**Review by vitaliiStefanchakInventorsoft — APPROVED** [review]

**Comment by vitaliiStefanchakInventorsoft** [comment]
ConsiderItDoneItsDoneGIF (2)

**Pull request closed without merging**